core/src/main/java/jenkins/model/lazy/AbstractLazyLoadRunMap.java

```diff
@@ -592,7 +592,7 @@ public synchronized int maxNumberOnDisk() {
         return numberOnDisk.max();
     }
 
-    protected final synchronized void proposeNewNumber(int number) throws IllegalStateException {
+    protected final synchronized void proposeNewNumber(int number) {
```

Comment:
[Effective Java 3rd Edition §74](https://books.google.com/books?id=BIpDDwAAQBAJ) says:

> Use the Javadoc `@throws` tag to document each exception that a method can throw, but do not use the `throws` keyword on unchecked exceptions.

This PR removes the `throws` keyword on this unchecked exception (good), but it does not add the Javadoc `@throws` tag (bad).